mlrun/builder.py

```diff
@@ -256,6 +256,16 @@ def configure_kaniko_ecr_init_container(kpod, registry, repo):
         # https://github.com/GoogleContainerTools/kaniko#pushing-to-amazon-ecr
         # we only need this in the kaniko container
         kpod.env.append(client.V1EnvVar(name="AWS_SDK_LOAD_CONFIG", value="true"))
+
+        # ensure "AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY" have no values
+        # as they might be volumized via a project secret and will conflict with the instance role
+        kpod.pod.spec.containers[0].env.extend(
+            [
+                client.V1EnvVar(name="AWS_ACCESS_KEY_ID", value=""),
+                client.V1EnvVar(name="AWS_SECRET_ACCESS_KEY", value=""),
+            ]
+        )
```

Comment:
Do these parameters need to be empty in the init-container or other containers? If they are needed in the init-container, then use the `init_container_env` dictionary defined earlier in this method instead. If not, then I wouldn't put this code in this function (which as its name implies configures the init-container...) but rather in the parent function that calls it.

Reply:
Seems like the function name doesn't  relate to what is really done in the function

Reply:
So changing the function name seems the way to go IMO

Reply:
Why not? It appends an init-container to the pod and configures it. It does configure some things that are pod-level, such as mounting volumes and stuff, but it does what it's called.

Reply:
To clarify - my issue is mostly with having to use `kpod.pod.spec.containers[0]` to access a different container than the one the function really deals with.

Reply:
moved logic to correct place